### PR TITLE
Update scala and pekko dependencies

### PR DESCRIPTION
**Pull request by nvollmar**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -12,24 +12,24 @@ import sbt._
...
 
-  val pekkoVersion = System.getProperty("override.pekko.version", "1.0.1")
+  val pekkoVersion = System.getProperty("override.pekko.version", "1.0.2")
   val pekkoVersionInDocs = "1.0"
   val cassandraVersionInDocs = "4.0"
 
   // Should be sync with the version of the driver in Pekko Connectors Cassandra
-  val driverVersion = "4.15.0"
-  val driverVersionInDocs = "4.14"
+  val driverVersion = "4.17.0"
```
Comment (pjfanning) [inline project/Dependencies.scala:24]: Can we put the cassandra driver update in a separate PR? I think the Scala and Pekko upgrades could be backported to 1.0.x branch.

Comment (nvollmar) [reply project/Dependencies.scala:24]: sure

Comment (nvollmar) [reply project/Dependencies.scala]: moved to #121

**Review by pjfanning — APPROVED** [review]
lgtm

**Review by mdedetrich — APPROVED** [review]
lgtm
